ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -806,6 +811,103 @@ func (ipfs *Connector) RepoStat(ctx context.Context) (*api.IPFSRepoStat, error)
 	return &stats, nil
 }
 
+// RepoGC performs a garbage collection sweep on the cluster peer's IPFS repo.
+func (ipfs *Connector) RepoGC(ctx context.Context) (*api.RepoGC, error) {
+	ctx, span := trace.StartSpan(ctx, "ipfsconn/ipfshttp/RepoGC")
+	defer span.End()
+
+	ctx1, cancel1 := context.WithTimeout(ctx, ipfs.config.IPFSRequestTimeout)
+	defer cancel1()
+
+	id := &api.ID{}
+	err := ipfs.rpcClient.CallContext(
```

Comment:
You can fill the peerID/peerName information of this object in `cluster.go` when you process the answers of the broadcast (like `globalPinInfoCid` does). Slightly simpler, but mostly for consistency.

Reply:
That would mean adding one more cluster method and rpc method say `RepoGCLocal`. So far I have directly called IPFSConnector's rpc call. 
But I should change this and add `RepoGCLocal`, since that's how everywhere we have handled `?local=true` option

Reply:
previous comment is unrelated to `local`. In general yes, copy from how `Status*` things are done. Previous comments just asks to save this rpc call since you can set that information later.

Reply:
In `globalPinInfoCid`, peername is set to the peer string. I don't see any way to get that accept calling `ID` (in cluster method or connector method). If skipping that rpc call is more important, there is no reason to keep peer name in repo gc struct, since it will be same as peer id.

Reply:
Ah yes, on error it sets the peer ID as peer name. But otherwise the pintracker has set it because it's initialized with the peername (and does not need to make an rpc). In any case, you can set the peername correctly in there directly too in cluster.go. (sorry, I thought globalPinInfo did the same).